core/templates/dev/head/domain/collection/CollectionRightsObjectFactorySpec.js

```diff
@@ -0,0 +1,80 @@
+// Copyright 2016 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Tests for CollectionRightsObjectFactory.
+ */
+
+describe('Collection rights object factory', function() {
+  var CollectionRightsObjectFactory = null;
+
+  beforeEach(module('oppia'));
+
+  beforeEach(inject(function($injector) {
+    CollectionRightsObjectFactory = $injector.get(
+      'CollectionRightsObjectFactory');
+  }));
+
+  it('should be able to set public', function() {
```

Comment:
Can users who do not have edit access publish a collection? If not, it might be worth updating the object and tests (and maybe add tests) to reflect this behavior. I think the code trying to publish a collection for a user without edit access is also an exceptional case (similar to any user trying to unpublish a collection when they don't have rights to do so).